contributor-docs/adrs/adr-005.md

```diff
@@ -0,0 +1,65 @@
+# ADR 5: Migration plan for deprecated components
+
+## Status
+
+Proposed
+
+## Context
+
+As we work on maturity of our components, we will sometimes need to deprecate components that are replaced with new components.
+
+## Here are the 3 proposed stages:
+
+### Stage 1
+
+Start new component outside the main bundle. Folks who want to try it, need to explicitly import it from the `drafts` bundle.
+
+`import { ActionMenu} from '@primer/react/drafts'`
+
+The contract with consumers is - you are opting into a rewrite of the old component that might not cover all the cases of the old component yet. If you are using both the old and new version of the component in different pages, you are paying some additional bundlesize cost.
+
+Note: If it's a 1:1 replacement, it's useful to keep the same name.
+
+### Stage 2
+
+After we have the confidence that this component is better than the old version of it, we swap the components and move the old component out of the main bundle.
```

Comment:
What factors determine how confident we are? If it's in `main`, it's already been code reviewed.

Reply:
Good question! I think it's when we think we are happy with the API and it have internally tested that it covers all the use cases of the previous component. 

For example, I'd like to ship all the Menu components that compose ActionList together, so I'm waiting till I finish SelectPanel